documentation/docs/examples/cad-2doverlay.mdx

```diff
@@ -13,118 +13,146 @@ that follows the object it's attached to. Note that these overlays always will b
+callback is triggered every time the HTML overlay is updated.
+
+```js runnable
+function createOverlay() {
+  const element = document.createElement('div');
+  element.innerText = `Overlay!`;
+  element.style.cssText = `
+    position: absolute;
+    
+    /* Anchor to the center of the element and ignore events */
+    transform: translate(-50%, -50%);
+    pointer-events: none;
+    touch-action: none;
+
+    /* Make it look nice */
+    padding: 10px;
+    color: #fff;
+    background: #232323da;
+    borderRadius: 0.25em;
+    border: '#ffffff22 solid 2px;
+  `;
+  return element;
 }
 
-// Handle mouse clicks
+const overlays = viewer.createHtmlOverlayTool();
 viewer.on('click', event => {
   // Find 3D coordinate of what the user clicks
   const intersection = viewer.getIntersectionFromPixel(event.offsetX, event.offsetY);
   if (intersection) {
-    updateOverlayPosition(intersection.point);
-    viewer.on('cameraChange', (_position, _target) => {
-      updateOverlayPosition(intersection.point);
-    });
+    const el = createOverlay();
+    const options = {
+      positionUpdatedCallback: (element, position2D, position3D, distanceToCamera) => {
+        // Compute opacity based on distance to camera
+        const opacity = 1.0 - (Math.min(Math.max(distanceToCamera, 10), 110) - 10) / 100.0;
+        element.style.opacity = opacity;
+      }
+    };
+    overlays.add(el, intersection.point, options);
   }
 });
 ```
 
-Now, the overlay position is updated whenever the camera is moved. Note that we also handle the case when the clicked position goes out of view by hiding the overlay.
+As shown above, overlays can be removed using `remove()`. It is also possible to remove all 
+overlays by disposing the tool:
+
+```js runnable
+function createOverlay() {
+  const element = document.createElement('div');
+  element.innerText = `!`;
+  element.style.cssText = `
+    position: absolute;
+    
+    /* Anchor to the center of the element and ignore events */
+    transform: translate(-50%, -50%);
+    pointer-events: none;
+    touch-action: none;
+
+    /* Make it look nice */
+    padding: 10px;
+    minHeight: 50px;
+    color: #000000;
+    background: #f5a614;
+    borderRadius: 0.25em;
+    border: '#ffffff22 solid 2px;
+  `;
+  return element;
+}
 
-<video preload="auto" autoPlay="autoplay" loop controls>
-  <source src={useBaseUrl('/img/cad-overlay-updating.webm')} />
-</video>
+const overlays = viewer.createHtmlOverlayTool();
+const bounds = model.getModelBoundingBox();
+for (let i = 0; i < 100; i++) {
+  const s = [Math.random(), Math.random(), Math.random()];
+  const pos = new THREE.Vector3(
+      bounds.min.x + (bounds.max.x - bounds.min.x) * s[0],
+      bounds.min.y + (bounds.max.y - bounds.min.y) * s[1],
+      bounds.min.z + (bounds.max.z - bounds.min.z) * s[2]
+    );
+  overlays.add(createOverlay(), pos);
+}
+
+// Dispose all overlays after 10 seconds
+setTimeout(() => overlays.dispose(), 10000);
```

Comment:
if the viewer got disposed would overlays be disposed too by the viewer? I think they should be. And I think it would be convenient if disposal would destroy all created HTML elements.